.github/workflows/build-and-test.yml

```diff
@@ -56,8 +59,14 @@ jobs:
         ) }}
 
     steps:
-      - name: check out
-        uses: actions/checkout@v1
+      - name: push check out
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/checkout@v2
+      - name: pull request check out
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Could we factor this into a reusable action? I think we can create a [composite run steps action](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-composite-run-steps-action) and [put that action inside `.github/actions/`](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/about-actions#choosing-a-location-for-your-action)

Reply:
I was thinking about doing that once I started working on the changes